src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -657,7 +653,16 @@ int SolverInterfaceImpl::setMeshVertex(
   mesh::PtrMesh mesh(context.mesh);
   PRECICE_DEBUG("MeshRequirement: {}", fmt::streamed(context.meshRequirement));
   index = mesh->createVertex(internalPosition).getID();
-  mesh->allocateDataValues();
+
+  mesh->allocateDataValues(); //@todo remove this call? But complicated, because write mapping expects already initialized toData
+
+  const auto newSize = mesh->verticesSize();
+  for (auto &context : _accessor->writeDataContexts()) {
+    if (context.getMeshName() == mesh->getName()) {
+      context.resizeBufferTo(newSize);
+    }
+  }
```

Comment:
The data on the mesh is needed because of the way the mapping is implemented. As soon as the mapping maps from storage to storage, we will not need `mesh->allocateDataValues()` anymore. I opened #1651 for this. Doing this here is out of scope, because `allocateDataValues()` is called very often in the tests and it also requires a general redesign of the mapping.